tmt/steps/__init__.py

```diff
@@ -989,7 +991,7 @@ def _method(cls: PluginClass) -> PluginClass:
         plugin_method = Method(name, class_=cls, doc=doc, order=order)
 
         # FIXME: make sure cls.__bases__[0] is really BasePlugin class
-        cast('BasePlugin', cls.__bases__[0])._supported_methods \
+        cast('BasePlugin[Any]', cls.__bases__[0])._supported_methods \
```

Comment:
Added a comment, `BasePlugin[Any]` stays - I suspect this needs more work since the function is used as a decorator, and annotating of decorators is sort of a WiP.